### ISSUE TITLE
add lint for prevent global import

### ISSUE THREAD

**Issue by little-dude** [rust-lang/rust-clippy#1228]
[Clippy already wans when importing all the variants of an enum](https://github.com/Manishearth/rust-clippy/wiki#enum_glob_use). But this is still allowed:

``` rust
use crate::*;
```

This kind of global import makes the code more difficult to read since we don't know where symbols come from. It may also confuse IDEs. Could we add a lint to prevent this?

[Link](https://www.reddit.com/r/rust/comments/52vkzc/examples_that_use_all_can_make_it_hard_to_learn) to the relevant reddit discussion

**Comment by oli-obk** [comment]
this should be a restriction lint.

one hard optional feature would be to figure out what is currently used and add a suggestion to replace it with a full import list.

**Comment by kennytm** [comment]
Note that prelude modules such as 

```
use iron::prelude::*;
```

may use a different lint ID, so one could allow glob-import of the preludes but warn against other glob imports.

**Comment by felix91gr** [comment]
I do not know the syntax by memory, so I have to ask: is this still an issue after the module changes made for Rust 2018 Edition?

**Comment by Manishearth** [comment]
Yep. Global imports weren't changed.

**Comment by mati865** [comment]
FWIW VS Code and Intellij IDEA both offer imports deglobbing which replace `*` with required imports.

**Comment by felix91gr** [comment]
> FWIW VS Code and Intellij IDEA both offer imports deglobbing which replace `*` with required imports.

That's super awesome :D

---

Is there a way to ask the compiler about the modules or symbols that the current file is using? If so, I think I know how to write this lint:

1. Check if there are any wildcard imports (I'm assuming this info is available to Clippy).
2. If there are, ask about the modules/symbols used by the current file.
3. Intersect (1) and (2)
4. If the result of (3) is not empty, lint about it!
